File: core/src/main/java/com/web3auth/core/types/MfaSettings.kt
package com.web3auth.core.types

import androidx.annotation.Keep

@Keep
data class MfaSettings(
    @Keep private var deviceShareFactor: MfaSetting? = null,
    @Keep private var backUpShareFactor: MfaSetting? = null,
    @Keep private var socialBackupFactor: MfaSetting? = null,
    @Keep private var passwordFactor: MfaSetting? = null,
    @Keep private var passkeysFactor: MfaSetting? = null,
    @Keep private var authenticatorFactor: MfaSetting? = null,
)

File: Assets/Plugins/Web3AuthSDK/Types/MfaSettings.cs
public class MfaSettings
{
    private MfaSetting deviceShareFactor;
    private MfaSetting backUpShareFactor;
    private MfaSetting socialBackupFactor;
    private MfaSetting passwordFactor;

    public MfaSetting DeviceShareFactor
    {
        get { return deviceShareFactor; }
        set { deviceShareFactor = value; }
    }

    public MfaSetting BackUpShareFactor
    {
        get { return backUpShareFactor; }
        set { backUpShareFactor = value; }
    }

    public MfaSetting SocialBackupFactor
    {
        get { return socialBackupFactor; }
        set { socialBackupFactor = value; }
    }

    public MfaSetting PasswordFactor
    {
        get { return passwordFactor; }
        set { passwordFactor = value; }
    }

    // Constructors
    public MfaSettings(
        MfaSetting deviceShareFactor = null,
        MfaSetting backUpShareFactor = null,
        MfaSetting socialBackupFactor = null,
        MfaSetting passwordFactor = null)
    {
        deviceShareFactor = deviceShareFactor;
        backUpShareFactor = backUpShareFactor;
        socialBackupFactor = socialBackupFactor;
        passwordFactor = passwordFactor;
    }
}